scipy/sparse/tests/test_base.py

```diff
@@ -3467,19 +3552,76 @@ def test_argmax(self):
+
         D1 = np.empty((0, 5))
         D2 = np.empty((5, 0))
+        explicit_values = [True, False]
 
-        for axis in [None, 0]:
+        for axis, explicit in itertools.product([None, 0], explicit_values):
             mat = self.spmatrix(D1)
-            assert_raises(ValueError, mat.argmax, axis=axis)
-            assert_raises(ValueError, mat.argmin, axis=axis)
+            assert_raises(ValueError, mat.argmax, axis=axis, explicit=explicit)
+            assert_raises(ValueError, mat.argmin, axis=axis, explicit=explicit)
 
-        for axis in [None, 1]:
+        for axis, explicit in itertools.product([None, 1], explicit_values):
             mat = self.spmatrix(D2)
-            assert_raises(ValueError, mat.argmax, axis=axis)
-            assert_raises(ValueError, mat.argmin, axis=axis)
+            assert_raises(ValueError, mat.argmax, axis=axis, explicit=explicit)
+            assert_raises(ValueError, mat.argmin, axis=axis, explicit=explicit)
 
+        # zero-size matrices
+        D = np.zeros((0, 10))
+        mat = self.spmatrix(D)
+        axes_even = [0, -2]
+        axes_odd = [1, -1]
+        even_explicit_pairs = list(itertools.product(axes_even,
+                                                     explicit_values))
+        odd_explicit_pairs = list(itertools.product(axes_odd,
+                                                    explicit_values))
+
+        for axis, explicit in even_explicit_pairs:
+            assert_raises(ValueError, mat.argmin, axis=axis, explicit=explicit)
+            assert_raises(ValueError, mat.argmax, axis=axis, explicit=explicit)
+        for axis, explicit in odd_explicit_pairs:
+            assert_array_equal(
+                np.zeros((0, 1)),
+                mat.argmin(axis=axis, explicit=explicit).A)
+            assert_array_equal(
+                np.zeros((0, 1)),
+                mat.argmax(axis=axis, explicit=explicit).A)
+
+        D = np.zeros((10, 0))
+        mat = self.spmatrix(D)
+
+        for axis, explicit in even_explicit_pairs:
+            mat.argmin(axis=axis, explicit=explicit)
+            assert_array_equal(
+                np.zeros((1, 0)),
+                mat.argmin(axis=axis, explicit=explicit).A)
+            assert_array_equal(
+                np.zeros((1, 0)),
+                mat.argmax(axis=axis, explicit=explicit).A)
+
+        for axis, explicit in odd_explicit_pairs:
+            assert_raises(ValueError, mat.argmin, axis=axis, explicit=explicit)
+            assert_raises(ValueError, mat.argmax, axis=axis, explicit=explicit)
+
+        D = np.zeros((2, 2))
+        mat = self.spmatrix(D)
+        if mat.nnz != 0:
+            # Noncanonical case
+            assert_equal(mat.argmin(axis=None, explicit=True), 0)
+            assert_equal(mat.argmax(axis=None, explicit=True), 0)
+        else:
+            assert_raises(ValueError, mat.argmin, axis=None, explicit=True)
+            assert_raises(ValueError, mat.argmax, axis=None, explicit=True)
```

Comment:
Just checking:  do you mean a breakdown of explicit rather than using even_explicit_pairs and odd_explicit_pairs ?
        for axis in axes_even:
            assert_raises(ValueError, mat.argmin, axis=axis, explicit=False)
            assert_raises(ValueError, mat.argmax, axis=axis, explicit=False)
            assert_raises(ValueError, mat.argmin, axis=axis, explicit=True)
            assert_raises(ValueError, mat.argmax, axis=axis, explicit=True)